Swiften/Parser/PayloadParsers/MIXJoinParser.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/Parser/PayloadParsers/MIXJoinParser.h>
+
+#include <boost/optional.hpp>
+
+#include <Swiften/Parser/PayloadParserFactory.h>
+#include <Swiften/Parser/PayloadParserFactoryCollection.h>
+#include <Swiften/Parser/PayloadParsers/MIXSubscribeParser.h>
+#include <Swiften/Parser/PayloadParsers/FormParser.h>
+
+using namespace Swift;
+
+MIXJoinParser::MIXJoinParser(PayloadParserFactoryCollection* parsers) : parsers_(parsers), level_(0) {
+}
+
+MIXJoinParser::~MIXJoinParser() {
+}
+
+void MIXJoinParser::handleStartElement(const std::string& element, const std::string& ns, const AttributeMap& attributes) {
+    if (level_ == 0) {
+        if (boost::optional<std::string> attributeValue = attributes.getAttributeValue("channel")) {
```

Comment:
As far as I'm aware, assignment in if-statements is a C++17 feature, and Swift is still at C++11.  Could you lift it out of the if, please?